client/components/gradeForm.jsx

```diff
@@ -0,0 +1,66 @@
+import React from 'react';
+
+class GradeForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      name: '',
+      course: '',
+      grade: 0
+    };
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  handleChange(event) {
+    const name = event.target.name;
+    const value = event.target.value;
+    this.setState({
+      [name]: value
+    });
+  }
+
+  handleSubmit(event) {
+    event.preventDefault();
+    this.props.addNewGrade({
+      name: this.state.name,
+      course: this.state.course,
+      grade: Number(this.state.grade)
+    });
+    this.handleReset();
+  }
+
+  handleReset() {
+    this.setState({
+      name: '',
+      course: '',
+      grade: 0
+    });
+  }
+
+  render() {
+    return (
+      <div className="form_container col-3">
+        <form onSubmit={this.handleSubmit}>
```

Comment:
You should also pass the reset handler to the form `onReset` so that the state gets updated if the user clicks the reset button.